src/test/groovy/transform/stc/GenericsSTCTest.groovy

```diff
@@ -1800,6 +1800,19 @@ assert result == 'ok'
 '''
     }
 
+    // GROOVY-7742
+    void testMethodParamWithDifferentPlaceholderForTypeArgument() {
+        // We use <T> here while List uses <E>
```

Comment:
The problem does not appear when using a different placeholder than `List` (i.e. not `E`), it appears when using the same as `Class` (`T`).